dbus-gridmeter_sml.py

```diff
@@ -112,51 +119,60 @@ def _getSmartMeterOverallConsumption(self):
         return value
 
 
-    def _getSmlSmartmeterData(self):
+    def _getSmlSmartmeterData(self, parse = False):
         try:
             start = time.time()
             sml_frame = None
             stream = SmlStreamReader()
 
             while sml_frame is None:
+              # we should get 1 msg per second, but sometimes it takes longer
+              if time.time()-start > 6:
```

Comment:
why so long timeout? wouldn't it be better to restart earlier? I think I had more good readings with the old solution. Anyhow I see your points, maybe one can fiddle a bit around to find optimal settings

Reply:
I actually had it sometimes that it would only give me every 2nd reading from the meter, and the other ones always timed out With this delay it's rock solid, not a single IR timeout over a whole day. And the normal update frequency still stays at 1 second.

Reply:
Initially, I had a lot of errors with your version:
```
2023-04-14 05:34:49.812842500 2023-04-14 05:34:49,811 root INFO Smartmeter IR timeout
2023-04-14 05:34:55.605950500 2023-04-14 05:34:55,604 root INFO Smartmeter IR timeout
2023-04-14 05:35:01.728695500 2023-04-14 05:35:01,720 root INFO Smartmeter IR timeout
2023-04-14 05:35:07.787961500 2023-04-14 05:35:07,786 root INFO Smartmeter IR timeout
2023-04-14 05:35:14.817927500 2023-04-14 05:35:14,816 root INFO Smartmeter IR timeout
2023-04-14 05:35:18.912756500 2023-04-14 05:35:18,911 root INFO Smartmeter IR timeout
2023-04-14 05:35:22.707855500 2023-04-14 05:35:22,705 root INFO Smartmeter IR timeout
2023-04-14 05:35:26.705265500 2023-04-14 05:35:26,704 root INFO Smartmeter IR timeout
2023-04-14 05:35:30.653207500 2023-04-14 05:35:30,651 root INFO Smartmeter IR timeout
2023-04-14 05:35:34.628740500 2023-04-14 05:35:34,627 root INFO Smartmeter IR timeout
2023-04-14 05:35:38.658006500 2023-04-14 05:35:38,656 root INFO Smartmeter IR timeout
2023-04-14 05:35:42.667175500 2023-04-14 05:35:42,657 root INFO Smartmeter IR timeout
2023-04-14 05:35:46.328933500 2023-04-14 05:35:46,327 root INFO Smartmeter IR timeout
2023-04-14 05:35:50.302114500 2023-04-14 05:35:50,300 root INFO Smartmeter IR timeout
2023-04-14 05:35:53.308891500 2023-04-14 05:35:53,307 root INFO Smartmeter IR timeout
2023-04-14 05:35:57.708140500 2023-04-14 05:35:57,706 root INFO Smartmeter IR timeout
2023-04-14 05:36:01.526360500 2023-04-14 05:36:01,525 root INFO Smartmeter IR timeout
```

and then also recognised that I had also such errors on another port:

```
2023-04-22 08:02:25.425231500     self._dbusservice = VeDbusService("{}.sml_{:02d}".format(servicename, deviceinstance))
2023-04-22 08:02:25.440266500   File "/opt/victronenergy/dbus-gridmeter_sml/vedbus.py", line 78, in __init__
2023-04-22 08:02:25.440275500     self._dbusname = dbus.service.BusName(servicename, self._dbusconn, do_not_queue=True)
2023-04-22 08:02:25.440280500   File "/usr/lib/python3.8/site-packages/dbus/service.py", line 144, in __new__
2023-04-22 08:02:25.440285500     raise NameExistsException(name)
2023-04-22 08:02:25.440288500 dbus.exceptions.NameExistsException: Bus name already exists: com.victronenergy.grid.sml_40
2023-04-22 08:32:04.719893500 2023-04-22 08:32:04,718 root INFO Start gridmeter_sml
2023-04-22 08:32:04.749110500 2023-04-22 08:32:04,739 root CRITICAL Error at main
2023-04-22 08:32:04.749118500 Traceback (most recent call last):
2023-04-22 08:32:04.749121500   File "/opt/victronenergy/dbus-gridmeter_sml/dbus-gridmeter_sml.py", line 292, in main
2023-04-22 08:32:04.749126500     pvac_output = DbusSmlSmartmeterService(
2023-04-22 08:32:04.749129500   File "/opt/victronenergy/dbus-gridmeter_sml/dbus-gridmeter_sml.py", line 31, in __init__
2023-04-22 08:32:04.749134500     self._dbusservice = VeDbusService("{}.sml_{:02d}".format(servicename, deviceinstance))
2023-04-22 08:32:04.749255500   File "/opt/victronenergy/dbus-gridmeter_sml/vedbus.py", line 78, in __init__
2023-04-22 08:32:04.749261500     self._dbusname = dbus.service.BusName(servicename, self._dbusconn, do_not_queue=True)
2023-04-22 08:32:04.749266500   File "/usr/lib/python3.8/site-packages/dbus/service.py", line 144, in __new__
2023-04-22 08:32:04.749270500     raise NameExistsException(name)
2023-04-22 08:32:04.749273500 dbus.exceptions.NameExistsException: Bus name already exists: com.victronenergy.grid.sml_40

```

and found with `ps` that there were concurrent processes running trying to grab a gridmeter.

From my other venus projects, I already learned, that the serial-starter probing doesn't work properly when you exit your code just with `exit(1)`. You have to use `sys.exit(1)`. exit() is just for the console.

please fix that then I can merge

Reply:
... and do you also want to integrate the serial number version conversion? just asking

Reply:
Ah thanks, I will fix that and also do the serial number conversion. Had a busy week but I think I'll have some time later today.